lib/Doctrine/Migrations/Configuration/Configuration.php

```diff
@@ -318,7 +320,7 @@ public function createMigrationTable() : bool
     public function getDateTime(string $version) : string
     {
         $datetime = str_replace('Version', '', $version);
-        $datetime = DateTime::createFromFormat('YmdHis', $datetime);
+        $datetime = DateTime::createFromFormat(self::VERSION_FORMAT, $datetime);
```

Comment:
Please use DateTimeImmutable.

Reply:
Should I be making unrelated changes in the same PR? I thought about this when I started but decided not to do it in this PR.

Reply:
Up to you here, but the other occurence is new code.

Reply:
I kept it the same as everything else so it is consistent in this sha. I will send another PR to change it everywhere.

Reply:
Created an issue for this here https://github.com/doctrine/migrations/issues/774